collector/prometheus.go

```diff
@@ -236,7 +239,7 @@ func (c *PrometheusCollector) getQueryRangeResults(queries *querys, results *map
 
 	if len(queriesToRetry) > 0 {
 		retries--
-		waitTime := time.Duration(5-retries) * time.Second
+		waitTime := time.Duration(math.Pow(2, float64(MaxRetries-retries))) * time.Second
```

Comment:
This might be an edge case, but if `retries` is greater than `MaxRetries`, the wait duration ends up being 0s until they are equal. 
```suggestion
                sleep := math.Max(math.Pow(2, float64(MaxRetries-retries), 1)
		waitTime := time.Duration(sleep) * time.Second
```

Reply:
done.